projects/portal/src/app/models/cosmos/staking.application.service.ts

```diff
@@ -270,6 +270,193 @@ export class StakingApplicationService {
+      }
+    }
+    const gas = { denom: 'uguu', amount: '1000000' };
+    const fee = null;
+    return { gas, fee };
+  }
+
+  async editValidatorSimple(
+    editValidatorData: EditValidatorData,
+    minimumGasPrice: proto.cosmos.base.v1beta1.ICoin,
+    privateKeyString: string,
+    gasRatio: number,
+    options?: InterfaceValidatorSimpleOptions,
+  ) {
+    const currentCosmosWallet = await this.walletService.currentCosmosWallet$
+      .pipe(take(1))
+      .toPromise();
+    if (!currentCosmosWallet) {
+      throw Error('Current connected wallet is invalid!');
+    }
+    const privateKey = convertHexStringToUint8Array(privateKeyString);
+    if (!privateKey) {
+      this.snackBar.open('Invalid PrivateKey!', 'Close');
+      return;
+    }
+
+    const dialogRef = this.loadingDialog.open('Sending Tx to be validator...');
+    const { disableRedirect, disableErrorSnackBar, disableSimulate } = options || {};
+
+    try {
+      const { gas, fee } = await this.handleSimulateToEditValidator(
+        editValidatorData,
+        minimumGasPrice,
+        gasRatio,
+        currentCosmosWallet.public_key,
+        disableSimulate,
+      );
+
+      const editValidatorResult = await this.staking.editValidator(
+        editValidatorData,
+        currentCosmosWallet,
+        gas,
+        fee,
+        privateKey,
+      );
+      const txHash = editValidatorResult.tx_response?.txhash;
+      if (txHash === undefined) {
+        throw Error('Invalid txHash!');
+      }
+      this.snackBar.open('Success', undefined, { duration: 6000 });
+      if (!disableRedirect) {
+        const redirectUrl =
+          location.port === '80' || location.port === '443' || location.port === ''
+            ? `${location.protocol}//${location.hostname}/explorer/validators/${editValidatorData.validator_address}`
+            : `${location.protocol}//${location.host}/explorer/validators/${editValidatorData.validator_address}`;
+        window.location.href = redirectUrl;
+      }
+      return txHash;
+    } catch (error) {
+      console.error(error);
+      if (!disableErrorSnackBar) {
+        this.snackBar.open(`Error: ${(error as Error).message}`, 'Close');
+      }
+      return;
+    } finally {
+      dialogRef.close();
+    }
+  }
+
+  async editValidator(
+    editValidatorData: EditValidatorData,
+    minimumGasPrice: proto.cosmos.base.v1beta1.ICoin,
+    gasRatio: number,
+    options?: InterfaceValidatorSimpleOptions,
+  ) {
+    const privateWallet: StoredWallet & { privateKey: string } =
+      await this.walletApplicationService.openUnunifiKeyFormDialog();
+    if (!privateWallet || !privateWallet.privateKey) {
+      this.snackBar.open('Failed to get Wallet info from dialog! Tray again!', 'Close');
+      return;
+    }
+
+    if (!validatePrivateStoredWallet(privateWallet)) {
+      this.snackBar.open('Invalid Wallet info!', 'Close');
+      return;
+    }
+
+    const currentCosmosWallet = await this.walletService.currentCosmosWallet$
```

Comment:
I made this `currentCosmosWallet` for common methods of `signTx` in `staking.editValidator()`
Is it OK? 
Should I somehow convent `privateWallet: StoredWallet` (Line 377) to `currentCosmosWallet`

Reply:
I used `convertStoredWalletToCosmosWallet` functon in  waller.service.ts